lib/spack/spack/ci.py

```diff
@@ -705,14 +732,49 @@ def generate_gitlab_ci_yaml(
             files (spack.yaml, spack.lock), etc should be written.  GitLab
             requires this to be within the project directory.
     """
+    rev1, rev2 = get_change_revisions()
+    tty.warn(f"Got following revisions: rev1={rev1}, rev2={rev2}")
+
+    changed = rev1 and rev2
+    if changed:
+        affected_pkgs = compute_affected_packages(rev1, rev2)
+        tty.debug("affected pkgs:")
+        for p in affected_pkgs:
+            tty.debug(f"  {p}")
+
+        possible_builds = spack.package_base.possible_dependencies(*env.user_specs)
+        # with env.read_transaction():
+        changed = any((spec in p for p in possible_builds.values()) for spec in affected_pkgs)
+
+    # Get the joined "ci" config with all of the current scopes resolved
+    ci_config = cfg.get("ci")
+
+    if not changed:
```

Comment:
@scottwittenburg When you do the next part of the refactor to create CI factories, it might be nice to be able to generate a noop pipeline directly, ignoring everything else.